# Sprint 10

La compañía de extracción de petróleo OilyGiant desea encontrar los mejores lugares donde abrir 200 pozos nuevos de petróleo en tres regiones diferentes.Se realizarán análisis estadísticos y de modelado predictivo para identificar la región más rentable y evaluar los riesgos asociados.

## Cargar liberías 

In [1]:
import pandas as pd
import numpy as np
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
import numpy as np

In [2]:
# Leer los archivos

df_0 = pd.read_csv('/datasets/geo_data_0.csv')
df_1 = pd.read_csv('/datasets/geo_data_1.csv')
df_2 = pd.read_csv('/datasets/geo_data_2.csv')

In [3]:
display(df_0.head())
display(df_1.head())
display(df_2.head())

print(df_0.info())
print()
print(df_1.info())
print()
print(df_2.info())

id        f0        f1        f2     product
0  txEyH  0.705745 -0.497823  1.221170  105.280062
1  2acmU  1.334711 -0.340164  4.365080   73.037750
2  409Wp  1.022732  0.151990  1.419926   85.265647
3  iJLyR -0.032172  0.139033  2.978566  168.620776
4  Xdl7t  1.988431  0.155413  4.751769  154.036647

id         f0         f1        f2     product
0  kBEdx -15.001348  -8.276000 -0.005876    3.179103
1  62mP7  14.272088  -3.475083  0.999183   26.953261
2  vyE1P   6.263187  -5.948386  5.001160  134.766305
3  KcrkZ -13.081196 -11.506057  4.999415  137.945408
4  AHL4O  12.702195  -8.147433  5.004363  134.766305

id        f0        f1        f2     product
0  fwXo0 -1.146987  0.963328 -0.828965   27.758673
1  WJtFt  0.262778  0.269839 -2.530187   56.069697
2  ovLUW  0.194587  0.289035 -5.586433   62.871910
3  q6cA6  2.236060 -0.553760  0.930038  114.572842
4  WPMUX -0.515993  1.716266  5.899011  149.600746

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column

In [4]:
print(df_0.isnull().sum())
print()
print(df_1.isnull().sum())
print()
print(df_2.isnull().sum())

id         0
f0         0
f1         0
f2         0
product    0
dtype: int64

id         0
f0         0
f1         0
f2         0
product    0
dtype: int64

id         0
f0         0
f1         0
f2         0
product    0
dtype: int64


Ninguno de los DF contiene valores ausentes.

El tipo de datos de los DataFrame corresponde a los valores de cada columna

## Entrena y prueba el modelo para cada región en geo_data_0.csv

### Divide los datos en un conjunto de entrenamiento y un conjunto de validación en una proporción de 75:25.


In [5]:
# División del conjunto de datos en dos: conjunto de entrenamiento (75%) y conjunto de validación (25%)

# Dividir los datos de las regiones

train_0, valid_0 = train_test_split(df_0, test_size=0.25, random_state=12345)
train_1, valid_1 = train_test_split(df_1, test_size=0.25, random_state=12345)
train_2, valid_2 = train_test_split(df_2, test_size=0.25, random_state=12345)

### Entrena el modelo y haz predicciones para el conjunto de validación.

In [6]:
# Entrenamiento del modelo y predicciones

def train_value_models(train, valid):
    features = ['f0', 'f1', 'f2']
    target = 'product'
    
    model = LinearRegression()
    model.fit(train[features], train[target])
    
    predictions = model.predict(valid[features])
    return predictions, valid[target]

### Guarda las predicciones y las respuestas correctas para el conjunto de validación.



In [7]:
# Se almacenan las predicciones y las respuestas reales (las actuales)

predictions_0, real_0 = train_value_models(train_0, valid_0)
predictions_1, real_1 = train_value_models(train_1, valid_1)
predictions_2, real_2 = train_value_models(train_2, valid_2)

results_0 = pd.DataFrame({'predictions': predictions_0, 'real': real_0})
results_1 = pd.DataFrame({'predictions': predictions_1, 'real': real_1})
results_2 = pd.DataFrame({'predictions': predictions_2, 'real': real_2})

### Muestra el volumen medio de reservas predicho y RMSE del modelo.

In [8]:
def calculate_metrics(results):
    mean_predictions = results['predictions'].mean()
    rmse = mean_squared_error(results['real'], results['predictions']) ** 0.5
    return mean_predictions, rmse

mean_predictions_0, rmse_0 = calculate_metrics(results_0)
mean_predictions_1, rmse_1 = calculate_metrics(results_1)
mean_predictions_2, rmse_2 = calculate_metrics(results_2)


print('Región 0')
print(f'volumen medio de reservas predicho: {mean_predictions_0} unidades')
print(f'RMSE: {rmse_0}')
print()
print('Región 1')
print(f'volumen medio de reservas predicho: {mean_predictions_1} unidades')
print(f'RMSE: {rmse_1}')
print()
print('Región 2')
print(f'volumen medio de reservas predicho: {mean_predictions_2} unidades')
print(f'RMSE: {rmse_2}')
print()

Región 0
volumen medio de reservas predicho: 92.59256778438035 unidades
RMSE: 37.5794217150813

Región 1
volumen medio de reservas predicho: 68.728546895446 unidades
RMSE: 0.893099286775617

Región 2
volumen medio de reservas predicho: 94.96504596800489 unidades
RMSE: 40.02970873393434



### Analiza los resultados

1. Volumen Medio de Reservas Predicho

El volumen medio de reservas predicho indica la cantidad promedio de reservas de petróleo que se espera encontrar en los pozos en cada región:

- Región 0: 92.59 unidades
- Región 1: 68.73 unidades
- Región 2: 94.97 unidades

La Región 2 tiene el mayor volumen medio de reservas predicho, seguida de la Región 0, mientras que la Región 1 tiene el volumen medio de reservas predicho más bajo.

2. Error Cuadrático Medio (RMSE)

La métrica de evaluación más comúnmente usada para las tareas de regresión es el error cuadrático medio o EMC. El EMC debe ser lo más bajo posible.

El RMSE mide la precisión del modelo, indicando cuánto se desvían, en promedio, las predicciones del modelo de los valores reales (actuales):

Región 0: 37.58
Región 1: 0.89
Región 2: 40.03

Un RMSE más bajo indica un modelo más preciso:

- Región 1: Tiene el RMSE más bajo (0.89), lo que sugiere que el modelo de regresión lineal predice con gran precisión el volumen de reservas en esta región.

- Región 0 y Región 2: Tienen RMSE significativamente más altos (37.58 y 40.03, respectivamente), lo que indica que las predicciones en estas regiones son menos precisas.

En general, 

Región 1 tiene el modelo de predicción más preciso, con un RMSE muy bajo (0.89). Sin embargo, el volumen medio de reservas predicho es el más bajo (68.73 unidades).

Región 2 tiene el mayor volumen medio de reservas predicho (94.97 unidades) pero también tiene un RMSE alto (40.03), lo que indica que las predicciones no son muy precisas.

Región 0 tiene un volumen medio de reservas predicho de 92.59 unidades y un RMSE de 37.58, lo que la coloca en un término medio en cuanto a volumen de reservas y precisión del modelo.

La Región 0 ofrece un equilibrio entre el volumen medio de reservas y la precisión del modelo, lo que podría ser una opción viable dependiendo de otros factores 

## Cálculo de ganancias:

Condiciones:

- Al explorar la región, se lleva a cabo un estudio de 500 puntos con la selección de los mejores 200 puntos para el cálculo del beneficio.
- El presupuesto para el desarrollo de 200 pozos petroleros es de 100 millones de dólares.
- Un barril de materias primas genera 4.5 USD de ingresos. El ingreso de una unidad de producto es de 4500 dólares (el volumen de reservas está expresado en miles de barriles).
- Después de la evaluación de riesgo, mantén solo las regiones con riesgo de pérdidas inferior al 2.5%. De las que se ajustan a los criterios, se debe seleccionar la región con el beneficio promedio más alto.


Dada la inversión de 100 millones por 200 pozos petrolíferos, de media un pozo petrolífero debe producir al menos un valor de 500,000 dólares en unidades para evitar pérdidas (esto es equivalente a 111.1 unidades). Compara esta cantidad con la cantidad media de reservas en cada región.

### Almacena todos los valores necesarios para los cálculos en variables separadas.

In [9]:
# Almacenar las variables
num_pozos = 200
ingre_unid_prod = 4500 # El ingreso de una unidad de producto es de 4.5 K USD
inversion = 100000000
costo_prod = 500000
cost_por_pozo = inversion / num_pozos
min_uni_pozo = 111.1

print(f'Cada pozo debe producir al menos {min_uni_pozo} unidades para evitar pérdidas.')

Cada pozo debe producir al menos 111.1 unidades para evitar pérdidas.


### Compara el umbral minimo de pérdidas con la cantidad media de reservas en cada región.

Dada la inversión de 100 millones por 200 pozos petrolíferos, de media un pozo petrolífero debe producir al menos un valor de 500,000 dólares en unidades para evitar pérdidas (esto es equivalente a 111.1 unidades). 

In [10]:
# Cantidad de media de reservas en cada región

mean_reserves_0 = df_0['product'].mean()
mean_reserves_1 = df_1['product'].mean()
mean_reserves_2 = df_2['product'].mean()

print(f'Promedio de reservas región 0: {mean_reserves_0} unidades')
print(f'Promedio de reservas región 1: {mean_reserves_1} unidades')
print(f'Promedio de reservas región 2: {mean_reserves_2} unidades')

Promedio de reservas región 0: 92.50000000000001 unidades
Promedio de reservas región 1: 68.82500000000002 unidades
Promedio de reservas región 2: 95.00000000000004 unidades


De media un pozo petrolífero debe producir al menos un valor de 500,000 dólares en unidades para evitar pérdidas (esto es equivalente a 111.1 unidades), en este caso el valro medio de unidades producido por cada región es:

- Región 0: 92.5
- Región 1: 68.825
- Región 2: 95

Todos estos valores están por debajo de la media del valor minimo para no producir pérdidas, lo que indica que en las 3 regiones la producción no alcanza los valores suficientemente altas para producir ganancias.

En conclusión ninguna de las tres regiones tiene una cantidad media de reservas que cumpla con la producción mínima requerida de 111.1 unidades para evitar pérdidas en promedio. Sin embargo, esto no necesariamente descarta ninguna región.

## Ganancia de un conjunto de pozos de petróleo seleccionados

### Elige los 200 pozos con los valores de predicción más altos de cada una de las 3 regiones (es decir, archivos 'csv')

In [11]:
def top_pozos(predictions, count=200):
    probs_sorted = predictions.sort_values(ascending=False)[:count]
    return probs_sorted

top_pozos_0 = top_pozos(pd.Series(results_0['predictions']))
top_pozos_1 = top_pozos(pd.Series(results_1['predictions']))
top_pozos_2 = top_pozos(pd.Series(results_2['predictions']))

### Resume el volumen objetivo de reservas según dichas predicciones. Almacena las predicciones para los 200 pozos para cada una de las 3 regiones.

In [12]:
sum_top_pozos_0 = top_pozos_0.sum()
sum_top_pozos_1 = top_pozos_1.sum()
sum_top_pozos_2 = top_pozos_2.sum()

print(f'Volumen total de los 200 mejores pozos de la región 0: {sum_top_pozos_0} unidades')
print(f'Volumen total de los 200 mejores pozos de la región 1: {sum_top_pozos_1} unidades')
print(f'Volumen total de los 200 mejores pozos de la región 2: {sum_top_pozos_2} unidades')

Volumen total de los 200 mejores pozos de la región 0: 31102.3308388114 unidades
Volumen total de los 200 mejores pozos de la región 1: 27746.026782163426 unidades
Volumen total de los 200 mejores pozos de la región 2: 29603.898658318347 unidades


### Calcula la ganancia potencial de los 200 pozos principales por región. Presenta tus conclusiones: propón una región para el desarrollo de pozos petrolíferos y justifica tu elección.

In [13]:
# Funcion de ganancias

def revenue(df_pozos, count= 200, cost_por_pozo= 500000):
    probs_sorted = df_pozos.sort_values(by='predictions', ascending=False)[:count]
    probs_real = probs_sorted['real']
    total_revenue = 4500 * probs_real.sum()
    total_cost = count * cost_por_pozo
    total_profit = total_revenue - total_cost
    return total_profit 

profit_0 = revenue(results_0)
profit_1 = revenue(results_1)
profit_2 = revenue(results_2)

# Calcular ganancias esperadas para cada región


print(f'Ganancia esperada región 0: ${profit_0}')
print(f'Ganancia esperada región 1: ${profit_1}')
print(f'Ganancia esperada región 2: ${profit_2}')


Ganancia esperada región 0: $33208260.43139851
Ganancia esperada región 1: $24150866.966815114
Ganancia esperada región 2: $27103499.635998324


In [14]:
# Región con mayor ganancia

max_profit = max(profit_0, profit_1, profit_2)

if max_profit == profit_0:
    print('La región 0 es la indicada para el desarrollo de pozos petroliferos')
elif max_profit == profit_1:
    print('La región 1 es la indicada para el desarrollo de pozos petroliferos')
else:
    print('La región 2 es la indicada para el desarrollo de pozos petroliferos')

La región 0 es la indicada para el desarrollo de pozos petroliferos


Ganancias de las 3 regiones:

- Región 0: $33,208,260.43

- Región 1: $24,150,866.96

- Región 2: $27,103,499.63

Volumen de las 3 regiones:

- Región 0: 31,102.33 Unidades
- Región 1: 27,746.02 Unidades
- Región 2: 29,603.89 Unidades


La mejor región para la producción de los pozos petroleros es la región 0, al tener una ganancia mayor en comparación con las otras regiones, siendo esta de $33,208,260.43 USD. Seguida de la región 2 y siendo la región 1, la región que produce menos ganancias en el desarrollo de pozos petroliferos.

El volumen total de los 200 mejores pozos también es mayor en la Región 0 con 31,102.33 unidades, lo que sugiere una mayor concentración de reservas en los mejores pozos.

La región 0 no solo tiene la mayor ganancia esperada, sino que también posee el mayor volumen total de reservas entre los 200 mejores pozos. Esto indica que, al elegir esta región, estamos seleccionando los pozos con el mayor rendimiento potencial.

La región 2 es la segunda mejor opción en términos de ganancias y volumen, pero aún está significativamente por debajo de la Región 0.

La región 1 presenta tanto la menor ganancia esperada como el menor volumen total de reservas entre los 200 mejores pozos, lo que la convierte en la opción menos favorable.

En conclusión general, la región indicada para el desarrollo de pozos petrolíferos es Región 0.

Esta región ofrece la mayor ganancia esperada y el mayor volumen de reservas en los pozos seleccionados, lo que maximiza el retorno de la inversión y reduce el riesgo de pérdidas.

## Calcula riesgos y ganancias para cada región

### Técnica del bootstrapping 

Utilizando las predicciones que almacenaste en el paso 4.2, emplea la técnica del bootstrapping con 1000 muestras para hallar la distribución de los beneficios.

In [15]:
def bootstrap_profit(df_region, n_bootstrap=1000, n_pozos=200, state=12345):
    state = np.random.RandomState(state)
    values = []
    for i in range(n_bootstrap):
        sample = df_region.sample(n=500, replace=True, random_state=state)
        profit = revenue(sample, n_pozos)
        values.append(profit)
    values = pd.Series(values)
    lower = values.quantile(0.025) # Intervalo de confianza del 95%
    upper = values.quantile(0.975)
    mean = values.mean()
    risk = (values < 0).mean() # Calcula el promedio de los valores de ganancia negativa (pérdida)
    loss_percentage = risk * 100  # convertir a porcentaje
    return mean, lower, upper, risk, loss_percentage

### Encuentra el beneficio promedio

Encuentra el beneficio promedio, el intervalo de confianza del 95% y el riesgo de pérdidas. La pérdida es una ganancia negativa, calcúlala como una probabilidad y luego exprésala como un porcentaje.

In [16]:
mean_profit_0, lower_0, upper_0, risk_0, loss_percentage_0 = bootstrap_profit(results_0)  
mean_profit_1, lower_1, upper_1, risk_1, loss_percentage_1 = bootstrap_profit(results_1)  
mean_profit_2, lower_2, upper_2, risk_2, loss_percentage_2 = bootstrap_profit(results_2)  

print('Región 0')
print('Ingresos promedios: $', mean_profit_0)
print(f'Intervalo de confianza 95%: \n Cuantil del 2.5%: ${lower_0} \n Cuantil del 97.5%: ${upper_0} \n Intervalo de confianza: ({upper_0}, {lower_0}) ({upper_0 - lower_0})')
print(f'Probabilidad de Riesgo de pérdida: {risk_0}')
print(f'Porcentaje de Riesgo de pérdida: {loss_percentage_0}%')
print()

print('Región 1')
print('Ingresos promedios: $', mean_profit_1)
print(f'Intervalo de confianza 95%: \n Cuantil del 2.5%: ${lower_1} \n Cuantil del 97.5%: ${upper_1} \n Intervalo de confianza: ({upper_1}, {lower_1}) ({upper_1 - lower_1})')
print(f'Probabilidad de Riesgo de pérdida: {risk_1}')
print(f'Porcentaje de Riesgo de pérdida: {loss_percentage_1}%')
print()

print('Región 2')
print('Ingresos promedios: $', mean_profit_2)
print(f'Intervalo de confianza 95%: \n Cuantil del 2.5%: ${lower_2} \n Cuantil del 97.5%: ${upper_2} \n Intervalo de confianza: ({upper_2}, {lower_2}) ({upper_2 - lower_2})')
print(f'Probabilidad de Riesgo de pérdida: {risk_2}')
print(f'Porcentaje de Riesgo de pérdida: {loss_percentage_2}%')

Región 0
Ingresos promedios: $ 3961649.8480237117
Intervalo de confianza 95%: 
 Cuantil del 2.5%: $-1112155.4589049604 
 Cuantil del 97.5%: $9097669.41553423 
 Intervalo de confianza: (9097669.41553423, -1112155.4589049604) (10209824.874439191)
Probabilidad de Riesgo de pérdida: 0.069
Porcentaje de Riesgo de pérdida: 6.9%

Región 1
Ingresos promedios: $ 4560451.057866608
Intervalo de confianza 95%: 
 Cuantil del 2.5%: $338205.0939898458 
 Cuantil del 97.5%: $8522894.538660347 
 Intervalo de confianza: (8522894.538660347, 338205.0939898458) (8184689.444670502)
Probabilidad de Riesgo de pérdida: 0.015
Porcentaje de Riesgo de pérdida: 1.5%

Región 2
Ingresos promedios: $ 4044038.665683568
Intervalo de confianza 95%: 
 Cuantil del 2.5%: $-1633504.1339559986 
 Cuantil del 97.5%: $9503595.749237997 
 Intervalo de confianza: (9503595.749237997, -1633504.1339559986) (11137099.883193996)
Probabilidad de Riesgo de pérdida: 0.076
Porcentaje de Riesgo de pérdida: 7.6%


### Presenta tus conclusiones:

Resultados obtenidos a través del método de bootstrapping de las 3 regiones, con un intervalo de confianza del 95%:

- Región 0

Ingresos promedios: $ 3,961,649.84

Intervalo de confianza 95%: 

 Cuantil del 2.5%: $-1,112,155.45
 
 Cuantil del 97.5%: $9,097,669.41 
 
 Intervalo de confianza: (9,097,669.41, -1,112,155.45) (10,209,824.87)
 
Probabilidad de Riesgo de pérdida: 0.069 (6.9%)


- Región 1

Ingresos promedios: $ 4,560,451.057

Intervalo de confianza 95%: 

 Cuantil del 2.5%: $338,205.093
 
 Cuantil del 97.5%: $8,522,894.538
 
 Intervalo de confianza: (8,522,894.53, 338,205.09) (8,184,689.44)
 
Probabilidad de Riesgo de pérdida: 0.015 (1.5%)
    

- Región 2

Ingresos promedios: $ 4,044,038.66

Intervalo de confianza 95%: 

 Cuantil del 2.5%: $-1,633,504.13
 
 Cuantil del 97.5%: $9,503,595.74
 
 Intervalo de confianza: (9,503,595.74, -1,633,504.13) (11,137,099.88)
 
Probabilidad de Riesgo de pérdida: 0.076

Porcentaje de Riesgo de pérdida: 7.6%


    

En el proyecto, los cuantiles fueron utilizados para determinar el intervalo de confianza del 95% de los beneficios esperados. 

Esto se hizo calculando el 2.5% y el 97.5% de los beneficios obtenidos mediante la técnica del bootstrapping, proporcionando una medida de la variabilidad y la incertidumbre en los beneficios esperados. Además, se utilizó para evaluar el riesgo de pérdidas, calculando la proporción de muestras en las que los beneficios eran negativos.

Calculamos el percentil 2.5 (P2.5) y el percentil 97.5 (P97.5) de las ganancias simuladas.

Este intervalo de confianza nos dice que estamos 95% seguros de que la ganancia real se encuentra entre estos dos valores.


Calculamos la probabilidad de que la ganancia sea negativa (pérdida) al ver qué porcentaje de las ganancias simuladas es menor que cero.

Percentil 2.5 (P2.5): Si este valor es muy bajo o negativo, indica que hay una probabilidad significativa de obtener ganancias bajas o pérdidas.

Percentil 97.5 (P97.5): Este valor indica el límite superior de las ganancias esperadas. Si es alto, muestra el potencial máximo de ganancias.

Riesgo de Pérdida: Un riesgo de pérdida bajo (<2.5%) indica que es poco probable que la inversión resulte en pérdidas, lo que es crucial para la toma de decisiones.


Análisis de Intervalos de Confianza

Región 0: Aunque su intervalo de confianza es más similar al de la región 1, la probabilidad de pérdidad es significativamente mayor siendo casi 6 veces mayor.

Región 1: El intervalo de confianza es más estrecho comparado con las otras regiones, además tiene un porcentaje de probabilidad de riesgo de pérdidas menor, estando dentro del rango establecido menor a 2.5%

Región 2: Tiene el intervalo de confianza más amplio y las pérdidas más altas.

**Región elegida**

Las tres regiones presentan ingresos promedio similares, pero probabilidades de riesgo de pérdida con diferencias. Sin embargo, de las tres, **la Región 1** con un ingreso promedio de $ 4,560,451.057 y una probabilidad de Riesgo de pérdida: 0.015 (1.5%)

## Conclusiones Generales

- Entrenamiento y Evaluación del Modelo:

La Región 1 mostró la mayor precisión en las predicciones con un RMSE de 0.89, aunque tuvo el menor volumen medio de reservas predicho (68.73 unidades).

La Región 2 tuvo el mayor volumen medio predicho (94.97 unidades), pero con la mayor imprecisión (RMSE de 40.03).

La Región 0 presentó un equilibrio entre el volumen medio de reservas (92.59 unidades) y la precisión del modelo (RMSE de 37.58).

- Cálculo de Ganancias:

Se determinó que cada pozo debe producir al menos 111.1 unidades para evitar pérdidas.

Ninguna región cumplió con este umbral en promedio

- Selección de Pozos:

Se calcularon las ganancias potenciales para los 200 mejores pozos:

Región 0: $33,208,260.43 USD

Región 1: $24,150,866.96 USD

Región 2: $27,103,499.63 USD

Conclusión: La Región 0 mostró la mayor ganancia esperada y el mayor volumen total de reservas entre los 200 mejores pozos.

- Evaluación de Riesgos y Ganancias con Bootstrapping:

**Región 0:**

Beneficio promedio: $3,961,649.85

Intervalo de confianza  (95%): [-$1,112,155.46, $9,097,669.42]

Riesgo de pérdida: 6.9%

**Región 1:**

Beneficio promedio: $4,560,451.06

Intervalo de confianza  (95%): [$338,205.09, $8,522,894.54]

Riesgo de pérdida: 1.5%

**Región 2:**

Beneficio promedio: $4,044,038.67

Intervalo de confianza  (95%): [-$1,633,504.13, $9,503,595.75]

Riesgo de pérdida: 7.6%


la Región 1 tiene el modelo de predicción más preciso y el menor riesgo de pérdida, aunque su volumen medio de reservas es el más bajo.

La Región 2 tiene el mayor volumen medio de reservas predicho, pero con una alta imprecisión y un mayor riesgo de pérdida.

La Región 0 ofrece un equilibrio entre el volumen medio de reservas y la precisión del modelo, pero con un riesgo de pérdida aceptable.


Región Recomendada:

La Región 1 es la opción más favorable para el desarrollo de pozos petrolíferos, dado su menor riesgo de pérdida y el beneficio promedio más alto.

Aunque la Región 0 tiene la mayor ganancia esperada, su riesgo de pérdida es mayor.

La Región 2 también es viable, pero tiene un mayor riesgo de pérdida comparado con la Región 1.

En conclusión, la **Región 1** es la indicada para el desarrollo de nuevos pozos petrolíferos, ya que maximiza el retorno de la inversión y minimiza el riesgo de pérdidas.